### PR TITLE
Bump AWS sdk 2.28.15

### PR DESCRIPTION
**Pull request by pditommaso**
This PR bump AWS SDK  2.28.15

### CODE REVIEW

**Comment by pditommaso** [comment]
@munishchouhan can you have a look at failing tests when you have chance?

**Comment by munishchouhan** [comment]
> @munishchouhan can you have a look at failing tests when you have chance?

ok sure

**Comment by pditommaso** [comment]
What the problem was?

**Comment by munishchouhan** [comment]
> What the problem was?

Still working on it, i think there is new check has been added, which is throwing NPE
```
java.lang.NullPointerException: Cannot invoke "software.amazon.awssdk.core.ClientEndpointProvider.isEndpointOverridden()" because "ep" is null
	at software.amazon.awssdk.core.interceptor.SdkExecutionAttribute.lambda$static$3(SdkExecutionAttribute.java:122)
	at software.amazon.awssdk.core.interceptor.ExecutionAttribute$DerivationValueStorage.lambda$set$0(ExecutionAttribute.java:277)
	at java.base/java.util.HashMap.compute(HashMap.java:1324)
	at software.amazon.awssdk.core.interceptor.ExecutionAttribute$DerivationValueStorage.set(ExecutionAttribute.java:277)
	at software.amazon.awssdk.core.interceptor.ExecutionAttributes.putAttribute(ExecutionAttributes.java:75)
	at software.amazon.awssdk.services.s3.internal.signing.DefaultS3Presigner.invokeInterceptorsAndCreateExecutionContext(DefaultS3Presigner.java:341)
	at software.amazon.awssdk.services.s3.internal.signing.DefaultS3Presigner.presign(DefaultS3Presigner.java:308)
	at software.amazon.awssdk.services.s3.internal.signing.DefaultS3Presigner.presignGetObject(DefaultS3Presigner.java:230)
	at io.seqera.wave.service.blob.signing.AwsS3BlobSigningService.createPresignedGetUrl(AwsS3BlobSigningService.groovy:79)
	at io.seqera.wave.service.blob.signing.AwsS3BlobSigningService.createSignedUri(AwsS3BlobSigningService.groovy:57)
	at io.seqera.wave.service.blob.impl.BlobCacheServiceImpl.blobDownloadUri(BlobCacheServiceImpl.groovy:216)
	at io.seqera.wave.service.blob.impl.BlobCacheServiceImplTest2.should create service(BlobCacheServiceImplTest2.groovy:62)
```

**Comment by munishchouhan** [comment]
I have also created a bug with aws sdk team about this
https://github.com/aws/aws-sdk-java-v2/issues/5646

**Comment by munishchouhan** [comment]
I think we are getting NPE from here

https://github.com/aws/aws-sdk-java-v2/blob/a9c5f266ceb1405ec17b898b287c782e5656ce80/core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/builder/AwsDefaultClientBuilder.java#L352

which is changed in this new version of SDK

**Comment by pditommaso** [comment]
Maybe it can be downgraded to a prior version. The goal here is to use a version that supports EKS pod identity. @gavinelder was mentioning `2.21.30` is enough

File: build.gradle [hunk 1]
```diff
@@ -66,9 +66,9 @@ dependencies {
     implementation("io.micronaut.cache:micronaut-cache-core")
     implementation("io.micronaut.cache:micronaut-cache-caffeine")
     implementation("io.micronaut.aws:micronaut-aws-parameter-store")
-    implementation "software.amazon.awssdk:ecr"
-    implementation "software.amazon.awssdk:ecrpublic"
-    implementation 'software.amazon.awssdk:ses'
+    implementation "software.amazon.awssdk:ecr:2.28.15"
+    implementation "software.amazon.awssdk:ecrpublic:2.28.15"
+    implementation 'software.amazon.awssdk:ses:2.28.15'
```
Comment (pditommaso) [inline build.gradle:71]:
```suggestion
    implementation "software.amazon.awssdk:ecr:2.21.30"
    implementation "software.amazon.awssdk:ecrpublic:2.21.30"
    implementation 'software.amazon.awssdk:ses:2.21.30'
```

**Comment by munishchouhan** [comment]
now tests are failing with this error:
```
'software.amazon.awssdk.awscore.internal.authcontext.AwsCredentialsAuthorizationStrategy$Builder software.amazon.awssdk.awscore.internal.authcontext.AwsCredentialsAuthorizationStrategy$Builder.defaultCredentialsProvider(software.amazon.awssdk.auth.credentials.AwsCredentialsProvider)'
java.lang.NoSuchMethodError: 'software.amazon.awssdk.awscore.internal.authcontext.AwsCredentialsAuthorizationStrategy$Builder software.amazon.awssdk.awscore.internal.authcontext.AwsCredentialsAuthorizationStrategy$Builder.defaultCredentialsProvider(software.amazon.awssdk.auth.credentials.AwsCredentialsProvider)'
	at software.amazon.awssdk.services.s3.internal.signing.DefaultS3Presigner.invokeInterceptorsAndCreateExecutionContext(DefaultS3Presigner.java:362)
	at software.amazon.awssdk.services.s3.internal.signing.DefaultS3Presigner.presign(DefaultS3Presigner.java:308)
	at software.amazon.awssdk.services.s3.internal.signing.DefaultS3Presigner.presignGetObject(DefaultS3Presigner.java:230)
	at io.seqera.wave.service.blob.signing.AwsS3BlobSigningService.createPresignedGetUrl(AwsS3BlobSigningService.groovy:79)
	at io.seqera.wave.service.blob.signing.AwsS3BlobSigningService.createSignedUri(AwsS3BlobSigningService.groovy:57)
	at io.seqera.wave.service.blob.impl.BlobCacheServiceImpl.blobDownloadUri(BlobCacheServiceImpl.groovy:216)
	at io.seqera.wave.service.blob.impl.BlobCacheServiceImplTest2.should create service(BlobCacheServiceImplTest2.groovy:62)
```

I will change the credential provider and try again

**Comment by munishchouhan** [comment]
This turn out to be dependency issue, when we are mentioning version, we need to have the whole sdk

**Comment by munishchouhan** [comment]
now failing to create shadowjar because of the files added by new dependency, i will try to enable `zip64 `
```
> shadow.org.apache.tools.zip.Zip64RequiredException: archive contains more than 65535 entries.
  
  To build this archive, please enable the zip64 extension.
  See: https://docs.gradle.org/8.8/dsl/org.gradle.api.tasks.bundling.Zip.html#org.gradle.api.tasks.bundling.Zip:zip64
```

**Comment by pditommaso** [comment]
> we need to have the whole sdk 

No, that's too big. Likely some other dep is missing

**Comment by munishchouhan** [comment]
> > we need to have the whole sdk
> 
> No, that's too big. Likely some other dep is missing

now I only included core and not the whole sdk

File: build.gradle [hunk 2]
```diff
@@ -204,3 +205,6 @@ jacocoTestReport {
         }))
     }
 }
+shadowJar {
```
Comment (pditommaso) [inline build.gradle:208]: Is this still needed ?

Comment (munishchouhan) [reply build.gradle:208]: yes

**Comment by pditommaso** [comment]
All sdk was still there, lets try this ☝️

**Comment by munishchouhan** [comment]
> All sdk was still there, lets try this ☝️

oops sorry, my bad

**Comment by pditommaso** [comment]
damn, still failing

**Comment by munishchouhan** [comment]
> damn, still failing

got the right one, fixed now

**Comment by pditommaso** [comment]
What a team work! 😆

**Comment by pditommaso** [comment]
Finally

**Comment by pditommaso** [comment]
This was reverted 3f197135 because aws sms was failing

**Pull request merged**